### PR TITLE
Add filters parameter to the S3Dataset constructor

### PR DESCRIPTION
**Pull request by zhenyu**
## Overview
To give the end user of S3Dataset more flexibility, this PR adds a filters parameters to the S3Dataset Constructor. 
The filters is passed to the Pyarrow parquet read_table, so follows the syntax of filters on https://arrow.apache.org/docs/python/generated/pyarrow.parquet.read_table.html.

## Notes
- This is backwards compatible because the `filters` parameter is set to None by default.

## Limits
- The syntax of the filters has been bound to Pyarrow which makes it harder to migrate to the other IO lib like [polars](https://pola.rs/), but since the wicker dataset already been [bound to the Pyarrow ecosystem](https://github.com/woven-planet/wicker/blob/main/wicker/core/datasets.py#L133), it should not be a problem in the shorterm
## Added

- Filters parameter in S3Dataset constructor
- test_filters_dataset in test_datasets 

## Testing
- CI

### CODE REVIEW

File: tests/test_datasets.py
Comment (marccarre) [inline tests/test_datasets.py:128]: Minor: typo `retrived` -> `retrieved`.

```suggestion
            retrieved_values_list = [ds[i]["foo"] for i in range(len(ds))]
            retrieved_values_list.sort()
            filtered_value_list.sort()
            self.assertListEqual(retrieved_values_list, filtered_value_list)
```

File: wicker/core/datasets.py
Comment (marccarre) [inline wicker/core/datasets.py:165]: Minor: use an affirmative sentence for lower "mental load" on the reader/user of the library, e.g.:

```suggestion
        :param filters: Only returns rows which match the filter. Defaults to None, i.e., returns all rows.
```

**Review by marccarre — APPROVED** [review]
A few minor suggestions for improvements, but LGTM otherwise!
(Approving with the assumption that you will address the suggestions made, in order to not block you. 😌 )

Thank you for this PR @zhenyu! 🙏🏻 😊

**Comment by zhenyu** [comment]
@marccarre Thanks so much for the caught, suggestions committed. 
@aalavian would you PTAL? Thanks a lot

**Review by aalavian — APPROVED** [review]
Code LGTM, Thanks @zhenyu , could we increase Wicker version before merging PR please?

**Comment by zhenyu** [comment]
> Code LGTM, Thanks @zhenyu , could we increase Wicker version before merging PR please?

@aalavian , sorry I merged before addressing your comments. Will bump the version when made a release.

**Pull request merged**
